plugins/debuginfo-install.py

```diff
@@ -74,59 +75,126 @@ def configure(self):
+
+        if self.debugsource_missing:
+            logger.info(_("Could not find debugsource packages for the following packages: %s") %
+                        ", ".join(sorted(self.debugsource_missing)))
 
         if errors_spec and self.base.conf.strict:
             raise dnf.exceptions.PackagesNotAvailableError(_("Unable to find a match"),
                                                            pkg_spec=' '.join(errors_spec))
 
-    def _dbg_available(self, dbgname, package, match_evra):
-        if match_evra:
-            return self.packages_available.filter(
-                name=dbgname,
-                epoch=int(package.epoch),
-                version=str(package.version),
-                release=str(package.release),
-                arch=str(package.arch))
-        else:
-            return self.packages_available.filter(
-                name=dbgname,
-                arch=str(package.arch))
-
-    def _di_install(self, package):
-        for dbgname in [package.debug_name, package.source_debug_name]:
-            if dbgname in self.dbgdone:
-                break
-            if self._dbg_available(dbgname, package, True):
-                di = "{0}-{1}:{2}-{3}.{4}".format(
-                    dbgname,
-                    package.epoch,
-                    package.version,
-                    package.release,
-                    package.arch)
-                self.base.install(di)
-            elif self._dbg_available(dbgname, package, False):
-                di = "{0}.{1}".format(dbgname, package.arch)
-                self.base.install(di)
-            else:
-                continue
-            self.dbgdone.append(dbgname)
-            break
-        else:
-            logger.info(
-                _("Could not find debuginfo for package: %s") % package)
+    def _install_debug_from_system(self, package, debug_name):
+        query = self.packages_available.filter(name=debug_name,
+                                               epoch=package.epoch,
+                                               version=package.version,
+                                               release=package.release,
+                                               arch=package.arch)
+
+        if query:
+            self._install(query)
+            return True
+
+        return False
+
+    def _install_debug(self, debug_name, base_nevra):
+        kwargs = {}
+
+        # if some part of EVRA was specified in the argument, add it as a filter
+        if base_nevra.epoch is not None:
+            kwargs["epoch__glob"] = base_nevra.epoch
+        if base_nevra.version is not None:
+            kwargs["version__glob"] = base_nevra.version
+        if base_nevra.release is not None:
+            kwargs["release__glob"] = base_nevra.release
+        if base_nevra.arch is not None:
+            kwargs["arch__glob"] = base_nevra.arch
+
+        query = self.packages_available.filter(name=debug_name, **kwargs)
```

Comment:
Please could you replace self.packages_available by self.base.query to prevent reinstalling installed packages